Automattic/wp-calypso
# Stats: Update PWYW flow with new slider

Related to #84859

## Proposed Changes

- Adds updated tier slider to PWYW flow.
- Refactors `TierUpgradeSlider` to be reusable across flows.
- Adds `StatsCommercialUpgradeSlider` and `StatsPWYWUpgradeSlider` for flow-specific logic.
- Updates Commercial flow to use new component.

<img width="598" alt="SCR-20231207-nwyj" src="https://github.com/Automattic/wp-calypso/assets/40267301/f6323c91-9067-4558-bea7-86fb4c79e588">

## Follow-on Changes

To come in follow-up PR(s):

- Remove the old `TierUpgradeSlider` component/file.
- Remove the old slider code from the PWYW flow. (it's hidden for now)
- Move the new `TierUpgradeSlider` to the component library.

## Testing Instructions



Currently behind the `stats/tier-upgrade-slider` flag. More to come…

1. Launch the Calypso Live link.
2. Navigate to **Stats → Traffic**.
3. Click on CTA to purchase Stats plan.
4. Add `stats/tier-upgrade-slider` to URL flags.
5. Confirm new tier slider is visible and updates the page correctly.

1. Click the "choose a commercial plan" link in the highlighted box.
2. Confirm the new tier slider shows correctly here too.

## Pre-merge Checklist
